extension/js/common/api/email-provider/gmail/google-auth.ts

```diff
@@ -200,72 +181,40 @@ export class GoogleAuth {
-        if (closedWinId === oauthWinId) {
-          chrome.windows.onRemoved.removeListener(onOauthWinClosed);
-          resolve({ result: 'Closed', acctEmail, id_token: undefined });
-        }
-      };
-      chrome.windows.onRemoved.addListener(onOauthWinClosed);
-    });
+  private static getParameterByName = (search: string, name: string) => {
+    const match = RegExp('[?&]' + name + '=([^&]*)').exec(search);
+    return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
   };
 
-  private static processOauthResTitle = (title: string): { result: GoogleAuthWindowResult$result, code?: string, error?: string, csrf?: string } => {
-    const parts = title.split(' ', 2);
-    const result = parts[0] as GoogleAuthWindowResult$result;
-    const params = Url.parse(['code', 'state', 'error'], parts[1]);
-    let authReq: AuthReq;
+  private static getAuthRes = async ({ acctEmail, save, authWindowResult }:
+    { acctEmail?: string, save: boolean, authWindowResult: Bm.AuthWindowResult }): Promise<AuthRes> => {
     try {
-      authReq = GoogleAuth.apiGoogleAuthStateUnpack(String(params.state));
-    } catch (e) {
-      return { result: 'Error', error: `Wrong oauth state response: ${e}` };
-    }
-    if (!['Success', 'Denied', 'Error'].includes(result)) {
-      return { result: 'Error', error: `Unknown google auth result '${result}'` };
-    }
-    return { result, code: params.code ? String(params.code) : undefined, error: params.error ? String(params.error) : undefined, csrf: authReq.csrfToken };
-  };
-
-  /**
-   * Is the title actually just url of the page? (means real title not loaded yet)
-   */
-  private static isUrl = (title: string) => {
-    return title.match(/^(?:https?:\/\/)?accounts\.google\.com/) !== null
-      || title.startsWith(GOOGLE_OAUTH_SCREEN_HOST.replace(/^https?:\/\//, ''))
-      || title.startsWith(GOOGLE_OAUTH_SCREEN_HOST);
-  };
-
-  private static isForwarding = (title: string) => {
-    return title.match(/^Forwarding /) !== null;
-  };
-
-  private static waitForAndProcessOauthWindowResult = async (windowId: number, acctEmail: string | undefined, scopes: string[], csrfToken: string, save: boolean): Promise<AuthRes> => {
-    while (true) {
-      const [oauth] = await tabsQuery({ windowId });
-      if (oauth?.title && oauth.title.includes(GoogleAuth.OAUTH.state_header) && !GoogleAuth.isUrl(oauth.title) && !GoogleAuth.isForwarding(oauth.title)) {
-        const { result, error, code, csrf } = GoogleAuth.processOauthResTitle(oauth.title);
-        if (error === 'access_denied') {
-          return { acctEmail, result: 'Denied', error, id_token: undefined }; // sometimes it was coming in as {"result":"Error","error":"access_denied"}
-        }
-        if (result === 'Success') {
-          if (!csrf || csrf !== csrfToken) {
-            return { acctEmail, result: 'Error', error: `Wrong oauth CSRF token. Please try again.`, id_token: undefined };
-          }
```

Comment:
Has the CSRF check moved elsewhere? I see CSRF token getting created with the request but don't see the check anymore, it may be vulnerable to CSRF attacks.

Reply:
Would be good to have a test for this, too. Since we use mocked tests, it should be straight forward. Could leave the test for another issue, but the checking needs to be fixed.

Reply:
UI test added